From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Tue, 19 Jun 2018 13:23:57 +0200
Subject: [PATCH] {MAYBE UPS} arm/firmware: disable coverage plugin when using
 raw registers

gcc fails to compile arch/arm/firmware/trusted_foundations.o:

    Error: .err encountered

This is because __asmeq("%0", "r0") is wrong when using the coverage
plugin. The generated assembly of function tf_generic_smc() looks like:

    tf_generic_smc:
    .fnstart
    @ Naked Function: prologue and epilogue provided by programmer.
    @ args = 0, pretend = 0, frame = 0
    @ frame_needed = 1, uses_anonymous_args = 0
    mov r4, r0 @ type, type
    mov r5, r1 @ arg1, arg1
    mov r6, r2 @ arg2, arg2
    bl __sanitizer_cov_trace_pc
    .syntax divided
    .arch_extension sec
    stmfd sp!, {r4 - r11, lr}
    .ifnc r4,r0; .ifnc r4r0,fpr11; .ifnc r4r0,r11fp; .ifnc r4r0,ipr12;
    .ifnc r4r0,r12ip; .err; .endif; .endif; .endif; .endif; .endif @ type

As r4 is not r0, it fails... Disable the plugin to make things right.
---
 arch/arm/firmware/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/firmware/Makefile b/arch/arm/firmware/Makefile
index a71f16536b6c..b801964d305d 100644
--- a/arch/arm/firmware/Makefile
+++ b/arch/arm/firmware/Makefile
@@ -1 +1,3 @@
+KCOV_INSTRUMENT_trusted_foundations.o := n
+
 obj-$(CONFIG_TRUSTED_FOUNDATIONS)	+= trusted_foundations.o
-- 
